server/src/test/java/org/apache/iotdb/db/integration/IoTDBUDTFBuiltinFunctionIT.java

```diff
@@ -295,4 +295,39 @@ public void testVariationTrendCalculationFunction(String functionName, double ex
       fail(throwable.getMessage());
     }
   }
+
+  @Test
+  public void testConstantTimeSeriesGeneratingFunctions() {
+    String[] expected = {
+      "0, 0.0, 0.0, 1024, 3.141592653589793, 2.718281828459045, ",
+      "2, 1.0, null, 1024, 3.141592653589793, 2.718281828459045, ",
+      "4, 2.0, null, 1024, 3.141592653589793, 2.718281828459045, ",
+      "6, null, 3.0, null, null, 2.718281828459045, ",
+      "8, null, 4.0, null, null, 2.718281828459045, ",
+    };
+
+    try (Connection connection =
+            DriverManager.getConnection(
+                Config.IOTDB_URL_PREFIX + "127.0.0.1:6667/", "root", "root");
+        Statement statement = connection.createStatement();
+        ResultSet resultSet =
+            statement.executeQuery(
+                "select s7, s8, const(s7, 'value'='1024', 'type'='INT64'), pi(s7, s7), e(s7, s8, s7, s8) from root.sg.d1")) {
+
+      assertEquals(1 + 5, resultSet.getMetaData().getColumnCount());
+
+      for (int i = 0; i < INSERTION_SQLS.length; ++i) {
+        resultSet.next();
+        StringBuilder actual = new StringBuilder();
+        for (int j = 0; j < 1 + 5; ++j) {
+          actual.append(resultSet.getString(1 + j)).append(", ");
+        }
+        assertEquals(expected[i], actual.toString());
+      }
+
+      assertFalse(resultSet.next());
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
```

Comment:
May add some more test cases.
1. Invalid queries to test the validate rule.
2. Wildcard paths as the UDF parameter.
3. Invalid paths as the UDF parameter.

And I'm not very sure whether the nested function query is available currently. If so, is this working well in complicated queries lessing IT cases to prove, or the following queries are not allowed by design? 

Such as `SELECT const(avg(s1), 'value'='1024', 'type'='int64')`, `SELECT const(sin(avg(s1)), 'value'='1024', 'type'='int64')` and so on.

Reply:
Thanks for your review!

> May add some more test cases.
> 1. Invalid queries to test the validate rule.

I will add tests to test the validation rule.

> 2. Wildcard paths as the UDF parameter.
> 3. Invalid paths as the UDF parameter.

These 2 are well tested in other ITs, so I think those tests are not necessary here.

> And I'm not very sure whether the nested function query is available currently. If so, is this working well in complicated queries lacking IT cases to prove, or the following queries are not allowed by design?

The nested expression (#4012) is available now! :D

Please see the user docs for more detail: https://github.com/apache/iotdb/blob/master/docs/UserGuide/IoTDB-SQL-Language/DML-Data-Manipulation-Language.md#nested-query-subquery

Functions of all kinds (multi inputs, single inputs, row-by-row iteration, window-by-window iteration, etc), arithmetic operators, and time series are proved to work well in nested expression by several ITs. The code coverage of the expression framework is about 90%. (See https://github.com/apache/iotdb/blob/master/server/src/test/java/org/apache/iotdb/db/integration/IoTDBNestedQueryIT.java)

So I think we don't need to test all functions in nested expressions. The only thing we need to do is to check the basic behavior of the function, which will simplify our work.

Reply:
> So I think we don't need to test all functions in nested expressions. The only thing we need to do is to check the basic behavior of the function, which will simplify our work.
Agreed.